public/pages/utils/anomalyResultUtils.ts

```diff
@@ -167,9 +175,272 @@ export const filterWithDateRange = (
+  let featureData = {} as { [key: string]: FeatureAggregationData[] };
+  rawAnomalies.forEach(item => {
+    if (get(item, 'top_anomaly_hits.hits.hits', []).length > 0) {
+      const rawAnomaly = get(item, 'top_anomaly_hits.hits.hits.0._source');
+      if (get(rawAnomaly, 'anomaly_grade') !== undefined) {
+        anomalies.push({
+          anomalyGrade: toFixedNumber(get(rawAnomaly, 'anomaly_grade')),
+          confidence: toFixedNumber(get(rawAnomaly, 'confidence')),
+          startTime: get(rawAnomaly, 'data_start_time'),
+          endTime: get(rawAnomaly, 'data_end_time'),
+          plotTime: get(rawAnomaly, 'data_end_time'),
+        });
+      }
+      if (get(rawAnomaly, 'feature_data', []).length > 0) {
+        get(rawAnomaly, 'feature_data', []).forEach(feature => {
+          if (!get(featureData, get(feature, 'feature_id'))) {
+            featureData[get(feature, 'feature_id')] = [];
+          }
+          featureData[get(feature, 'feature_id')].push({
+            data: toFixedNumber(get(feature, 'data')),
+            startTime: get(rawAnomaly, 'data_start_time'),
+            endTime: get(rawAnomaly, 'data_end_time'),
+            plotTime: get(rawAnomaly, 'data_end_time'),
+          });
+        });
+      }
+    }
   });
+  return {
+    anomalies: anomalies,
+    featureData: featureData,
+  };
+};
+
+export const parseAnomalySummary = (
+  anomalySummaryResult: any
+): AnomalySummary => {
+  const anomalyCount = get(
+    anomalySummaryResult,
+    'data.response.aggregations.count_anomalies.value',
+    0
+  );
+  return {
+    anomalyOccurrence: anomalyCount,
+    minAnomalyGrade: anomalyCount
+      ? toFixedNumber(
+          get(
+            anomalySummaryResult,
+            'data.response.aggregations.min_anomaly_grade.value'
+          )
+        )
+      : 0,
+    maxAnomalyGrade: anomalyCount
+      ? toFixedNumber(
+          get(
+            anomalySummaryResult,
+            'data.response.aggregations.max_anomaly_grade.value'
+          )
+        )
+      : 0,
+    minConfidence: anomalyCount
+      ? toFixedNumber(
+          get(
+            anomalySummaryResult,
+            'data.response.aggregations.min_confidence.value'
+          )
+        )
+      : 0,
+    maxConfidence: anomalyCount
+      ? toFixedNumber(
+          get(
+            anomalySummaryResult,
+            'data.response.aggregations.max_confidence.value'
+          )
+        )
+      : 0,
+    lastAnomalyOccurrence: anomalyCount
+      ? minuteDateFormatter(
+          get(
+            anomalySummaryResult,
+            'data.response.aggregations.max_data_end_time.value'
+          )
+        )
+      : '',
+  };
+};
+
+export const parsePureAnomalies = (
+  anomalySummaryResult: any
+): AnomalyData[] => {
+  const anomaliesHits = get(
+    anomalySummaryResult,
+    'data.response.hits.hits',
+    []
+  );
+  const anomalies = [] as AnomalyData[];
+  if (anomaliesHits.length > 0) {
```

Comment:
is this check needed?

Reply:
Yes, to make sure we have got anomaly results rather than empty result.

Reply:
I think you may just do `anomaliesHits.forEach` without checking the length > 0. small thing. nvm.